tests/api.c

```diff
@@ -41601,6 +41601,266 @@ static int test_wolfSSL_CTX_set_ecdh_auto(void)
     (void)ctx;
     return ret;
 }
+
+#if defined(OPENSSL_EXTRA) && defined(WOLFSSL_ERROR_CODE_OPENSSL)
+static THREAD_RETURN WOLFSSL_THREAD SSL_read_test_server_thread(void* args)
+{
+    callback_functions* callbacks = NULL;
+    WOLFSSL_CTX* ctx = NULL;
+    WOLFSSL*     ssl = NULL;
+    SOCKET_T     sfd = 0;
+    SOCKET_T     cfd = 0;
+    word16       port;
+    char msg[] = "I hear you fa shizzle!";
+    int  len   = (int) XSTRLEN(msg);
+    char input[1024];
+    int  idx;
+    int  ret, err;
+
+    if (!args)
+        return 0;
+
+    ((func_args*)args)->return_code = TEST_FAIL;
+
+    callbacks   = ((func_args*)args)->callbacks;
+    ctx         = wolfSSL_CTX_new(callbacks->method());
+
+#if defined(USE_WINDOWS_API)
+    port = ((func_args*)args)->signal->port;
+#else
+    /* Let tcp_listen assign port */
+    port = 0;
+#endif
+
+#ifdef WOLFSSL_TIRTOS
+    fdOpenSession(Task_self());
+#endif
+
+    AssertIntEQ(WOLFSSL_SUCCESS,
+        wolfSSL_CTX_load_verify_locations(ctx, caCertFile, 0));
+
+    AssertIntEQ(WOLFSSL_SUCCESS,
+        wolfSSL_CTX_use_certificate_file(ctx, svrCertFile,
+                                        WOLFSSL_FILETYPE_PEM));
+
+    AssertIntEQ(WOLFSSL_SUCCESS,
+        wolfSSL_CTX_use_PrivateKey_file(ctx, svrKeyFile,
+                                        WOLFSSL_FILETYPE_PEM));
+
+    if (callbacks->ctx_ready)
+        callbacks->ctx_ready(ctx);
+
+    ssl = wolfSSL_new(ctx);
+    AssertNotNull(ssl);
+
+    /* listen and accept */
+    tcp_accept(&sfd, &cfd, (func_args*)args, port, 0, 0, 0, 0, 1);
+    CloseSocket(sfd);
+
+    AssertIntEQ(WOLFSSL_SUCCESS, wolfSSL_set_fd(ssl, cfd));
+
+    if (callbacks->ssl_ready)
+        callbacks->ssl_ready(ssl);
+
+    do {
+        err = 0; /* Reset error */
+        ret = wolfSSL_accept(ssl);
+        if (ret != WOLFSSL_SUCCESS) {
+            err = wolfSSL_get_error(ssl, 0);
+        }
+    } while (ret != WOLFSSL_SUCCESS && err == WC_PENDING_E);
+
+    if (ret != WOLFSSL_SUCCESS) {
+        wolfSSL_free(ssl);
+        wolfSSL_CTX_free(ctx);
+        CloseSocket(cfd);
+        ((func_args*)args)->return_code = TEST_FAIL;
+        return 0;
+    }
+
+    /* read and write data */
+    XMEMSET( input, 0, sizeof(input));
+    idx = 0;
+    while (1) {
+        ret = wolfSSL_read(ssl, input + idx, sizeof(input));
+        if (ret > 0) {
+            idx += ret;
+            break;
+        }
+        else {
+            err = wolfSSL_get_error(ssl,ret);
+            if (err == WOLFSSL_ERROR_WANT_READ) {
+                continue;
+            }
+            break;
+        }
+    }
+
+    if (err == WOLFSSL_ERROR_ZERO_RETURN || err == WOLFSSL_ERROR_WANT_READ) {
```

Comment:
Why would err be WOLFSSL_ERROR_WANT_READ?
We continue the loop above if that was the case.

Reply:
Removed WOLFSSL_ERROR_WANT_READ from this if statement.